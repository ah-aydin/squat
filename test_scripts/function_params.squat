func func0() {
    println("This is ", func0);
}

func func1(int param1) {
    println("This is ", func0, "with param:", param1);
}

func sumup(int param1, int param2) int {
    return param1 + param2;
}

func call_a_func_with_param_int(func(int) fn, int param) {
    println("Calling call_a_func_with_param_int");
    fn(param);
}

func call_a_func_with_param_int_int_return_int(int param1, int param2, func(int, int) int fn) int {
    println("Calling function with 2 int and return int param");
    int value = fn(param1, param2);
    println("The value is", value, ". Returning value times 2");
    return value * 2;
}

func main() {
    println("hello there");
    func0();
    func1(10);
    int s;
    s = sumup(10, sumup(1, 2));
    println("The sum of the values are", s);

    int m = 10 < 10 ? 1 : 2;
    println("var m is", m, type(m));

    func(int, int) int very_important_func = sumup;
    var meh = very_important_func(10, 90);
    println("The very important function sum is", meh);

    call_a_func_with_param_int(func1, 1234);
    println("The returned value is:", call_a_func_with_param_int_int_return_int(99, 99, sumup));
}
