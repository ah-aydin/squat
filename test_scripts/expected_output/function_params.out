Compiling and running file: test_scripts/function_params.squat
hello there
This is <func func0>
This is <func func0> with param: 10
The sum of the values are13
var m is 2<type Int>
The very important function sum is 100
Calling call_a_func_with_param_int
Calling func <func func1> with param 124357
This is <func func0> with param: 125591
The returned value is: Calling function with 2 int and return int param
The value is 198. Returning value times 2
396
hello there
This is <func func0>
This is <func func0> with param: 10
The sum of the values are 13
var m is: 2<type Int>
The very important function sum is: 100
Calling call_a_func_with_param_int
Calling func <func func1> with param 124357
This is <func func0> with param: 125591
The returned value is: Calling function with 2 int and return int param
The value is 198. Returning value times 2
396
